src/Parallel/ResourceInfo.hpp

```diff
@@ -903,11 +906,15 @@ void ResourceInfo<Metavariables>::build_singleton_map(
 
     // Print some diagnostic info to stdout for each singleton. This can aid in
     // debugging.
-    Parallel::printf("Allocating %s on proc %d, exclusive = %s\n",
-                     pretty_type::name<component>(), *singleton_map.second,
-                     singleton_map.first ? "true" : "false");
+    ss << pretty_type::name<component>();
+    ss << " on node " << Parallel::node_of<int>(*singleton_map.second, cache);
+    ss << ", global proc " << *singleton_map.second;
+    ss << ", exclusive = " << (singleton_map.first ? "true" : "false") << "\n";
   });
 
+  ss << "\n";
+  Parallel::printf(ss.str());
```

Comment:
`Parallel::printf("%s", ss.str());` (Prevents problems if the string contains a %.)